composer/loggers/progress_bar_logger.py

```diff
@@ -56,10 +65,21 @@ def __init__(
             unit=unit,
         )
 
-    def log_data(self, data: Dict[str, Any]):
-        formatted_data = {k: format_log_data_value(v) for (k, v) in data.items() if k in self.keys_to_log}
-        self.metrics.update(formatted_data)
-        self.pbar.set_postfix(self.metrics)
+    def log_data(self, state: State):
+        dataloader_label = state.dataloader_label
+        if self.metric_string is None:
+            formatted_data = {}
+            curr_metrics = state.current_metrics.get(dataloader_label, {})
```

Comment:
Could we unify the logic between the if and the else -- e.g. `if metric_string is None: metric_string = ', '.join('k={k}' for k in curr_metrics)`, and then use the same logic for both cases?

Reply:
I don't think that would reduce the complexity much because of the default keys I need to add, like `'loss_train`' or the default accuracy